lib/md4.c

```diff
@@ -36,12 +36,17 @@
 #endif /* USE_OPENSSL */
 
 #ifdef USE_MBEDTLS
-#include <mbedtls/config.h>
 #include <mbedtls/version.h>
+#if MBEDTLS_VERSION_MAJOR < 3
```

Comment:
Since the change that altered the header includes was introduced in 3, the check should be reversed and include the new file iff we are compiling against 3 and not the other way around. Nitpick to some extent, but it will improve readability.

Also, why doesn't this (and the remaining changes) use `MBEDTLS_VERSION_NUMBER` as the rest if the code does?